app/views/contexts/preview_context.py

```diff
@@ -30,6 +34,8 @@ def __init__(
         )
         self._routing_path = None
         self.questionnaire_store = questionnaire_store
+        if not schema.json.get("preview_questions"):
```

Comment:
Since you use this key multiple times, I suggest making it a property called `preview_enabled` on schema.

Reply:
This can be done before the `__init__` as that isn't important if it isn't enabled.